packages/frontend/src/pages/campaigns/index.tsx

```diff
@@ -35,16 +39,28 @@ export const Campaigns = () => {
 
     useDebounce(() => setDebouncedSearchQuery(searchQuery), 300, [searchQuery]);
 
-    const { loading, kpiTokens: searchedResolvedKPITokens } =
-        useSearchedResolvedKPITokens(debouncedSearchQuery);
-
-    //  toggle filters
-    const toggleFilters = () => setFilterOpen(!filtersOpen);
-    const [filtersOpen, setFilterOpen] = useState(false);
-
     // page settings
-    const [pageSize, setPageSize] = useState(12);
+    const [filtersOpen, setFilterOpen] = useState(false);
+    const toggleFilters = () => setFilterOpen(!filtersOpen);
     const [ordering, setOrdering] = useState<SelectOption>(ORDERING_OPTIONS[0]);
+    const [currentPage, setCurrentPage] = useState(1);
+    const { data, totalPages } = usePagination(results, currentPage, 4);
+
+    const handlePageChange = (pageNumber: SetStateAction<number>) => {
```

Comment:
We should wrap this with a `useCallback` since the function gets passed down to `Pagination` and then `PaginationBall`.

Reply:
I'll add it. Thanks

Reply:
fixed: c3f356ab938dc7946f877253851d47318df5df6b